src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -10005,6 +9990,30 @@ void MethodTableBuilder::CheckForSystemTypes()
         {
             pMT->SetInternalCorElementType (ELEMENT_TYPE_I);
         }
+        else if ((strcmp(name, g_Int128Name) == 0) || (strcmp(name, g_UInt128Name) == 0))
+        {
+            EEClassLayoutInfo* pLayout = pClass->GetLayoutInfo();
+            pLayout->SetIsInt128OrHasInt128Fields(TRUE);
+#ifdef TARGET_ARM
+            // No such type exists for the Procedure Call Standard for ARM. We will default
+            // to the same alignment as __m128, which is supported by the ABI.
+
+            pLayout->m_ManagedLargestAlignmentRequirementOfAllMembers = 8;
+#elif defined(TARGET_64BIT) || defined(TARGET_X86)
+
+            // These types correspond to fundamental data types in the underlying ABIs:
+            // * Int128:  __int128
+            // * UInt128: unsigned __int128
+            //
+            // This behavior matches the ABI standard on various Unix platforms
+            // On Windows, no standard for Int128 has been established yet, 
+            // although applying 16 byte alignment is consistent with treatment of 128 bit SSE types
+            // even on X86
```

Comment:
Worth noting that the MSVC STL does have a private `__int128` that is 16-byte aligned. Clang also uses 16-byte alignment for its own `Int128` when targeting the Windows ABI.

Nothing official/documented of course, but it does help justify using `16` on Windows here for managed.